src/systems/a320_systems/src/hydraulic/mod.rs

```diff
@@ -4932,44 +4944,220 @@ impl SimulationElement for ElevatorSystemHydraulicController {
+
+    id_emergency_reset_1: Option<VariableIdentifier>,
+    id_emergency_reset_2: Option<VariableIdentifier>,
+
+    commanded_position: [Angle; 2],
+    active_mode: [bool; 2],
+    is_emergency_reset: bool,
+
+    component_type: RudderComponent,
+}
+impl A320RudderTrimTravelLimiterController {
+    fn new(context: &mut InitContext, component_type: RudderComponent) -> Self {
+        Self {
+            id_active_mode_1: context
+                .get_identifier(format!("RUDDER_{}_1_ACTIVE_MODE_COMMANDED", component_type)),
+            id_active_mode_2: context
+                .get_identifier(format!("RUDDER_{}_2_ACTIVE_MODE_COMMANDED", component_type)),
+            id_commanded_position_1: context
+                .get_identifier(format!("RUDDER_{}_1_COMMANDED_POSITION", component_type)),
+            id_commanded_position_2: context
+                .get_identifier(format!("RUDDER_{}_2_COMMANDED_POSITION", component_type)),
+
+            id_emergency_reset_1: if component_type == RudderComponent::Limiter {
+                Some(context.get_identifier("FAC_1_RTL_EMER_RESET".to_owned()))
+            } else {
+                None
+            },
+            id_emergency_reset_2: if component_type == RudderComponent::Limiter {
+                Some(context.get_identifier("FAC_2_RTL_EMER_RESET".to_owned()))
+            } else {
+                None
+            },
+
+            commanded_position: [Angle::default(); 2],
+            active_mode: [false; 2],
+            is_emergency_reset: false,
+
+            component_type,
+        }
+    }
+}
+impl AngularPositioningController for A320RudderTrimTravelLimiterController {
+    fn commanded_position(&self) -> [Angle; 2] {
+        self.commanded_position
+    }
+
+    fn energised_motor(&self) -> [bool; 2] {
+        self.active_mode
+    }
+
+    fn emergency_reset_active(&self) -> bool {
+        self.is_emergency_reset
+    }
+}
+impl SimulationElement for A320RudderTrimTravelLimiterController {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.commanded_position[0] = reader.read(&self.id_commanded_position_1);
+        self.commanded_position[1] = reader.read(&self.id_commanded_position_2);
+
+        // Trim commands from FBW is negative for right positive for left, opposite convention in hydraulics
+        if self.component_type == RudderComponent::Trim {
+            self.commanded_position[0] = -self.commanded_position[0];
+            self.commanded_position[1] = -self.commanded_position[1];
+        }
+
+        self.active_mode[0] = reader.read(&self.id_active_mode_1);
+        self.active_mode[1] = reader.read(&self.id_active_mode_2);
+
+        // Only limiter has an emergency reset system
+        if self.component_type == RudderComponent::Limiter {
+            self.is_emergency_reset = reader.read(&self.id_emergency_reset_1.unwrap())
+                || reader.read(&self.id_emergency_reset_2.unwrap());
+        }
```

Comment:
Reading this I actually have no clue what it's doing :D  
I'd prefer to keep the code more pleasant for new people coming in, because someone seeing this would just run away as fast as he can imho :D

Reply:
Ok. Was just a suggestion ;)